test/fixtures/es-module-loaders/mock-loader.mjs

```diff
@@ -192,48 +59,40 @@ export async function resolve(specifier, context, defaultResolve) {
 }
 
 export async function load(url, context, defaultLoad) {
-  doDrainPort();
-  if (url === 'node:mock') {
-    /**
-     * Simply grab the import.meta.doMock to establish the communication
-     * channel with preloadCode
-     */
-    return {
-      shortCircuit: true,
-      source: 'export default import.meta.doMock',
-      format: 'module'
-    };
-  }
   /**
    * Mocked fake module, not going to be handled in default way so it
    * generates the source text, then short circuits
    */
   if (url.startsWith('mock-facade:')) {
-    let [proto, version, encodedTargetURL] = url.split(':');
-    let ret = generateModule(mockedModuleExports.get(
-      decodeURIComponent(encodedTargetURL)
-    ));
+    let [_proto, _version, encodedTargetURL] = url.split(':');
+    let source = generateModule(encodedTargetURL);
     return {
       shortCircuit: true,
-      source: ret,
+      source,
       format: 'module'
     };
   }
   return defaultLoad(url, context);
 }
 
+const mainImportURL = new URL('./mock.mjs', import.meta.url);
```

Comment:
nit: it would be more elegant to have `mock.mjs` pass its `import.meta.url` to `initialize`, so renaming the file won’t make the test fail.